### ISSUE TITLE
Potentially Crucial Custom Rules Bug

### ISSUE THREAD

**Issue by tinder-cfuller** [realm/SwiftLint#1558]
# Issue Description

Disabling just one custom rule, actually disables all custom rules.

# STR

### 1) SwiftLint Config

Create config file named `.swiftlint.yml` with the following contents:

```yaml
custom_rules:
  hello_rule:
    regex: "Hello"
    match_kinds: string
  world_rule:
    regex: "World"
    match_kinds: string
```

### 2) Swift Test File

Create test file named `Example.swift` with the following contents:

```swift
class Example {

    func example() -> String {
        return "Hello World!"
    }

}
```

### 3) Run SwiftLint

When running SwiftLint, we successfully get the following output:

```
Loading configuration from '.swiftlint.yml'
Linting Swift files in current working directory
Linting 'example.swift' (1/1)
/PATH_OBSCURED/example.swift:4:17: warning: hello_rule Violation: Regex matched. (hello_rule)
/PATH_OBSCURED/example.swift:4:23: warning: world_rule Violation: Regex matched. (world_rule)
Done linting! Found 2 violations, 0 serious in 1 file.
```

As you can see ☝️ both rules are violated.

### 4) Disable One Rule

In the test file, disable only the `world_rule`:

```swift
class Example {

    // swiftlint:disable world_rule
    func example() -> String {
        return "Hello World!"
    }

}
```

### 5) Run SwiftLint Again

When running SwiftLint now, we get the following output:

```
Loading configuration from '.swiftlint.yml'
Linting Swift files in current working directory
Linting 'example.swift' (1/1)
Done linting! Found 0 violations, 0 serious in 1 file.
```

As you can see ☝️ there are no violations, when in fact, the `hello_rule` violation should be reported.

## Additional Info

This has been tested with various scenarios and the experience is consistently that disabling one custom rule, actually disables all custom rules. This also occurs with `:previous`, `:this`, `:next`.

This issue appears to only affects custom rules, not built-in rules.

**Comment by tinder-cfuller** [comment]
Test files…

[swiftlint-custom-rules-bug.zip](https://github.com/realm/SwiftLint/files/1023897/swiftlint-custom-rules-bug.zip)

**Comment by jpsim** [comment]
Thanks for reporting this. Filed a PR fixing this in #1603.

**Comment by tinder-cfuller** [comment]
@jpsim Awesomeness 👍
